src/deploymentmanager.ts

```diff
@@ -343,6 +375,79 @@ export class DeploymentManager implements IDeploymentManager {
         });
     }
 
+    private setupKeyvaultParameters(answers: Answers, outputs: any) {
+        this._keyVaultParams.solutionName.value = answers.solutionName;
+
+        const answerParams = ['aadTenantId', 
+                              'userPrincipalObjectId', 
+                              'solutionName',
+                              'aadTenantId',
+                              'appId',
+                              'servicePrincipalSecret',
+                              'deploymentId'];
+        answerParams.forEach((paramName) => {
+            if (this._keyVaultParams[paramName] !== undefined && answers[paramName] !== undefined) {
```

Comment:
May be just `if (this._keyVaultParams[paramName] && answers[paramName])` enough?

Reply:
Agreed this is a bit safer.